src/stochastic_triple.jl

```diff
@@ -154,6 +154,26 @@ end
 
 stochastic_triple(p::Real; kwargs...) = stochastic_triple(x -> x, p; kwargs...)
 
+function stochastic_triple_1(f, p::V, backend) where {V}
+    StochasticTriple{Tag{typeof(f), V}}(p, one(p), backend)
+end
+
+function stochastic_triple_0(f, p::V, backend) where {V}
+    StochasticTriple{Tag{typeof(f), V}}(p, zero(p), backend)
+end
+
+"""
+    stochastic_triple_vec(f, p::AbstractVector, ix; kwargs...)
+
+Generate a vector of stochastic_triple objects, where the element of `p` indexed at `ix` has 
+`δ` set to 1, and `δ` at all other elements is zero. This is useful for finding the components
+(partial derivatives) of the gradient of the function `f` at `p`.
+"""
+function stochastic_triple_vec(f, p::AbstractVector, ix; backend = PrunedFIs)
+    [i == ix ? stochastic_triple_1(f, p[i], backend) :
+     stochastic_triple_0(f, p[i], backend) for i in eachindex(p)]
+end
```

Comment:
Are `stochastic_triple_0` and `stochastic_triple_1` for performance reasons (in the sense of being function barriers, see https://docs.julialang.org/en/v1/manual/performance-tips/#kernel-functions)?